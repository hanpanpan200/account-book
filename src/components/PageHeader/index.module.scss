@import "../../styles/colors";

.container {
  display: flex;
  flex-direction: column;
  padding: 20px 16px 0;
  background-color: $red;
  .titleContainer {
    img {
      position: absolute;
      left: 10px;
      top: 42px;
      width: 24px;
      height: 24px;
    }
    h1 {
      color: $white;
      text-align: center;
      font-size: 20px;
      font-weight: 500;
      line-height: 40px;
    }
  }
}