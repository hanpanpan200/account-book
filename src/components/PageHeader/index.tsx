import React from 'react'
import classNames from 'classnames';

import styles from './index.module.scss'

const cx = classNames.bind(styles);

type Props = {
  title: string
  children?: JSX.Element[] | JSX.Element
  style?: string
}

const PageHeader: React.FC<Props> = ({ title, children, style }: Props) => {
  return (
    <header className={cx(styles.container, style)}>
      <h1 className={styles.title}>{title}</h1>
      {children}
    </header>
  )
}

export default PageHeader
