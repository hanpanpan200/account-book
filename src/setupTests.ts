// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { fireEvent } from '@testing-library/react';

export const fireClickEvent = (element: Document | Element | Window | Node) => {
  fireEvent(element, new MouseEvent('click', {
    bubbles: true,
    cancelable: true
  }));
}