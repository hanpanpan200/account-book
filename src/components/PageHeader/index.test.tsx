import React from 'react';
import { render, screen } from '@testing-library/react';
import { fireClickEvent } from 'setupTests';
import PageHeader from './index';

const mockedHistory = {
  goBack: jest.fn()
}

jest.mock('react-router-dom', () => {
  return {
    useHistory: () => mockedHistory
  }
})

describe('PageHeader', () => {
  it('should render correctly when show back button', () => {
    const { container } = render(
      <PageHeader title='HeaderTitle' showBackButton>
        <div>Header children</div>
      </PageHeader>
    );
    expect(container).toMatchSnapshot();
  });
  it('should navigate back when back button is clicked', () => {
    render(
      <PageHeader title='HeaderTitle' showBackButton>
        <div>Header children</div>
      </PageHeader>
    );
    let imgIcon = screen.getByAltText(/arrowLeftIcon/i);
    fireClickEvent(imgIcon);
    expect(mockedHistory.goBack).toBeCalled();
  });
  it('should render correctly when not show back button', () => {
    const { container } = render(
      <PageHeader title='HeaderTitle' showBackButton={false}>
        <div>Header children</div>
      </PageHeader>
    );
    expect(container).toMatchSnapshot();
  });
});


