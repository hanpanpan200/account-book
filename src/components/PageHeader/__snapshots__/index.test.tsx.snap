// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader should render correctly when not show back button 1`] = `
<div>
  <header
    class="container"
  >
    <div
      class="titleContainer"
    >
      <h1>
        HeaderTitle
      </h1>
    </div>
    <div>
      Header children
    </div>
  </header>
</div>
`;

exports[`PageHeader should render correctly when show back button 1`] = `
<div>
  <header
    class="container"
  >
    <div
      class="titleContainer"
    >
      <img
        alt="arrowLeftIcon"
        src="arrow-left.svg"
      />
      <h1>
        HeaderTitle
      </h1>
    </div>
    <div>
      Header children
    </div>
  </header>
</div>
`;
